### PR TITLE
Escape single quotes in metric labels in get_metrics()

### PR DESCRIPTION
**Pull request by JakeC-MyT**
This is a:
- [x] bug fix PR with no breaking changes
- [ ] new functionality

## Link to Issue 
https://github.com/dbt-labs/dbt-project-evaluator/issues/278

## Description & motivation
As discussed in the issue, currently the `get_metrics` macro can't handle single quotes in the label of a metric, causing a failure when building `stg_metrics`. This code adds a statement which sets `label` to be the single-quote escaped version of the label, and then passes that to the `wrap_string_with_quotes` function instead.

## Integration Test Screenshot
image

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [x] Snowflake
    - [ ] Databricks
    - [ ] DuckDB
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

File: macros/unpack/get_metrics.sql
Comment (dave-connors-3) [inline macros/unpack/get_metrics.sql:37]:
```suggestion
            wrap_string_with_quotes(dbt.escape_single_quotes(node.label))
```

**Review by dave-connors-3 — CHANGES_REQUESTED** [review]
Thanks @JakeC-MyT so much for the issue and PR! This is a good straightforward fix. 

I added a suggestion to make this a one-line change rather than creating a new jinja variable in the macro.

Additionally, I would love to have this case covered in our integration tests! I think the simplest way to do that would be to edit the existing metric in `integration_tests/models/marts/metrics.yml` to include a single quote in the label, and ensure our CI checks pass!

File: macros/unpack/get_metrics.sql
Comment (JakeC-MyT) [reply macros/unpack/get_metrics.sql:37]: Thanks @dave-connors-3 - totally bizarre I'm sure I tried this one line fix the first time around and it didn't work, but running it again now it looks fine! 

Thank you for the advice on the integration test - sorry I'm not too familiar with them so wasn't sure how best to demonstrate it works, will add a new commit now :)

File: integration_tests/models/marts/metrics.yml
Comment (dave-connors-3) [inline integration_tests/models/marts/metrics.yml:29]: was this a net new addition? not sure why this is flagged as a change here

**Review by dave-connors-3 — APPROVED** [review]
This looks great! thanks so much for the contribution! will merge once we get one more approval

File: integration_tests/models/marts/metrics.yml
Comment (JakeC-MyT) [reply integration_tests/models/marts/metrics.yml:29]: Not net new no - seems to be showing as a deletion followed by replacement. Sorry, may have got my branch in a bit of a tizz - I can try and resolve if you need, but don't think it has any difference from what's in main. Let me know, and thanks for your support in contributing - very happy to be involved!

**Review by graciegoheen — APPROVED** [review]
Nice work!

**Pull request merged**
